### PR TITLE
Feature/add manual definition input

### PR DESCRIPTION
**Pull request by justjohnd**
Made the following changes based on the design below:
1. Front and Back textarea fields should allow for manual entry and display in real time on the Preview card below
2. "Save | Clear" options should appear as soon as anything is entered into the front of the card
3. Pushing "Auto-Generate" should make an API call and populate the back of the card with information (or flash an error message)
4. Pushing "Save" should save the item to the "Deck" component below. (Currently, user will only see the number of cards saved so far).

- "Decks" input field and "Set" checkbox have been added, but don't have functionality yet.
- No substantial tests have been written for the component yet.

image

### CODE REVIEW

File: yambo-front/src/components/CardCreate.tsx [hunk 1]
```diff
@@ -1,21 +1,28 @@
 import React, { FC, ChangeEvent, MouseEvent, useState } from "react";
 import styled from "styled-components";
+import { TailSpin } from "react-loader-spinner";
```
Comment (mwhitman189) [inline yambo-front/src/components/CardCreate.tsx:3]: This does not appear to be in package.json. I think you either installed it globally, or installed without the "--save" flag, which is causing the test to fail.

File: package.json [hunk 2]
```diff
@@ -16,5 +16,8 @@
   "bugs": {
     "url": "https://github.com/mwhitman189/yambo-flashcards/issues"
   },
-  "homepage": "https://github.com/mwhitman189/yambo-flashcards#readme"
+  "homepage": "https://github.com/mwhitman189/yambo-flashcards#readme",
+  "dependencies": {
+    "react-loader-spinner": "^5.1.5"
+  }
 }
```
Comment (mwhitman189) [inline package.json:23]: There's no new spinner library in package.json.
Try `npm i react-loader-spinner --save` to add it to "depenencies"

File: yambo-front/src/components/CardCreate.tsx [hunk 3]
```diff
@@ -249,13 +292,15 @@ const CardCreate: FC = () => {
...
 
-  function handleSubmit(e: MouseEvent) {
+  function handleLookup(e: MouseEvent) {
     e.preventDefault();
 
+    setLoader(true);
+
     try {
       //Use for actual api call
       const fetchData = async () => {
```
Comment (mwhitman189) [inline yambo-front/src/components/CardCreate.tsx:311]: After the fetch call, you can check whether or not the response returned with a status of 200:
        if (response.status === 200) {
then continue setting the card data. And you could move the spinner outside of the         if (response.status === 200) {} so you only have to call setLoader(false) once.

**Review by mwhitman189 — CHANGES_REQUESTED** [review]
You got it all looking and working according to the specs. If you just address the issues here I'll merge the changes. Thanks!

**Comment by justjohnd** [comment]
- Add react-loader-spinner
- Adjust position of "Word Not Found" alert
- Modify "back" tab disable functionality. Not tab is only disabled until user clicks in either of the textareas
- Make "Front" textarea required

**Review by mwhitman189 — APPROVED** [review]

**Pull request merged**